plugins/discovery/manager.go

```diff
@@ -99,3 +116,25 @@ func (dm *Manager) Pause() {
 	log.Infof("stopping service discovery. new leader: %s", leadership.Leader())
 	dm.serviceListener.stop()
 }
+
+// resyncConfig periodically checks for changes to the discovery config.
+// It stops monitoring existing resources and reloads the discovery manager on changes
+func (dm *Manager) resyncConfig() {
+	if !dm.runConfig.DiscoveryConfig.EnableRuntimePlugins {
+		log.Info("runtime plugins disabled")
+		return
+	}
+
+	interval := dm.runConfig.DiscoveryConfig.DiscoveryInterval
+	log.Infof("discovery config interval: %v", interval)
+
+	go wait.Until(func() {
```

Comment:
'go wait.Until' looks very weird... why not just a ticker&select?

Reply:
It's similar. But more concise with additional error handling internally.